app/assets/javascripts/components/GlobalNav.js

```diff
@@ -0,0 +1,160 @@
+    this.delay = 250; // ms
+    this.mobileAnimationDuration = this.$globalNav.css('transition-duration');
+  };
+
+  /**
+  * Inherit from base module, for shared methods and interface
+  */
+  DoughBaseComponent.extend(GlobalNav);
+  GlobalNav.componentName = 'GlobalNav';
+
+  /**
+  * Initialize the component
+  */
+  GlobalNav.prototype.init = function(initialised) {
+    this._initialisedSuccess(initialised);
+    this._setUpMobileInteraction();
+    this._setUpDesktopInteraction();
+    this._setUpMobileAnimation();
+
+    $(window).on('resize', utilities.debounce($.proxy(this._resize, this), 100));
+
+    return this;
+  };
+
+  // Ensure mobile nav not activated on resize
+  GlobalNav.prototype._resize = function() {
+    this._setUpMobileAnimation();
+  }
+
+  GlobalNav.prototype._setUpMobileAnimation = function() {
+    if (mediaQueries.atSmallViewport()) {
+      this.$globalNav.css('transition-duration', this.mobileAnimationDuration);
+    } else {
+      this.$globalNav.css('transition-duration', '0s');
+    }
+  }
+
+  /**
+  * Set up events for mobile nav
+  */
+  GlobalNav.prototype._setUpMobileInteraction = function() {
+    var self = this;
+
+    this.$mobileNavButton.click(function() {
+      if (self.$globalNav.hasClass('is-active')) {
+        self._closeMobileNav();
+      } else {
+        self._openMobileNav();
+      }
+    });
+
+    this.$globalNavClumpHeading.click(function() {
+      self._openMobileSubNav(this);
+    });
+
+    this.$globalSubNavHeading.click(function() {
+      self._closeMobileSubNav(this);
+    })
+  }
+
+  /**
+  * Set up events for desktop nav
+  */
+  GlobalNav.prototype._setUpDesktopInteraction = function() {
+    var self = this;
+
+    // heading behaviour
+    this.$globalNavClumpHeading.not('.global-nav__clump__blog-link')
+      .mouseenter(function() {
+        self._openDesktopSubNav(this);
+      })
+      .mouseleave(function() {
+        self._closeDesktopSubNav(this);
+      });
+
+    // subnav behaviour
+    $('.global-subnav')
+      .mouseenter(function() {
+        self._openDesktopSubNav(this);
+      })
+      .mouseleave(function() {
+        self._closeDesktopSubNav(this);
+      });
+  }
+
+  GlobalNav.prototype._openMobileNav = function() {
+    this.$globalNav.addClass('is-active');
+  }
+
+  GlobalNav.prototype._closeMobileNav = function() {
+    this.$globalNav.removeClass('is-active');
+
+    if (this.$globalNavClumps.hasClass('is-active')) {
+      this.$globalNavClumps.removeClass('is-active');
+    }
+  }
+
+  GlobalNav.prototype._openMobileSubNav = function(index) {
+    // clear active clumps
+    this.$globalNavClump.removeClass('is-active');
+    this.$globalNavClumps.removeClass('is-active');
+
+    $(index)
+      .parents('[data-dough-nav-clump]').addClass('is-active')
+      .parents('[data-dough-nav-clumps]').addClass('is-active');
+  }
+
+  GlobalNav.prototype._closeMobileSubNav = function(index) {
```

Comment:
These look like a good candidate to be combined into a `_toggleMobileSubNav` function